commands/FBTextInputCommands.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python
+
+# Copyright (c) 2016, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+import os
+
+import lldb
+import fblldbbase as fb
+import fblldbviewhelpers as viewHelpers
+
+ACCESSIBILITY_ID = 0
+REPLACEMENT_TEXT = 1
+INPUT_TEXT = 0
+
+
+def lldbcommands():
+  return [
+    FBInputTexByAccessibilityIdCommand(),
+    FBInputTexToFirstResponderCommand(),
+  ]
+
+
+class FBInputTexByAccessibilityIdCommand(fb.FBCommand):
+  def name(self):
+    return 'setxt'
+
+  def description(self):
+    return 'Set text on text on a view by accessibility id.'
+
+  def args(self):
+    return [
+      fb.FBCommandArgument(arg='accessibilityId', type='string', help='The accessibility ID of the input view.'),
+      fb.FBCommandArgument(arg='replacementText', type='string', help='The text to set.')
+    ]
+
+  def run(self, arguments, options):
+    self.findView(rootView(), arguments[ACCESSIBILITY_ID], arguments[REPLACEMENT_TEXT])
+
+  def findView(self, view, searchIdentifier, replacementText):
+    views = subviewsOfView(view)
+    for index in range(0, viewsCount(views)):
+        subview = subviewAtIndex(views, index)
+        self.findView(subview, searchIdentifier, replacementText)
+    else:
+      identifier = accessibilityIdentifier(view)
+      if isEqualToString(identifier, searchIdentifier):
+        setTextInView(view, replacementText)
+
+
+class FBInputTexToFirstResponderCommand(fb.FBCommand):
+  def name(self):
+    return 'input'
+
+  def description(self):
+    return 'Input text into text field or text view that is first responder.'
+
+  def args(self):
+    return [
+      fb.FBCommandArgument(arg='inputText', type='string', help='The text to input.')
+    ]
+
+  def run(self, arguments, options):
+    self.findFirstResponder(rootView(), arguments[INPUT_TEXT])
+
+  def findFirstResponder(self, view, replacementText):
+    views = subviewsOfView(view)
+    for index in range(0, viewsCount(views)):
+        subview = subviewAtIndex(views, index)
+        self.findFirstResponder(subview, replacementText)
+    else:
+      if isFirstResponder(view):
+        setTextInView(view, replacementText)
```

Comment:
> this branch of code is only executed when a view has no subviews.

Right, and I'm wondering about cases where some text input view has internal subviews. There's no guarantee that that a text input view will always have no subviews. I think the `isFirstResponder` check should go above the loop, and if it fails, then loop over subviews. This would make it breadth first, but I don't see a problem with that, do you?